### ISSUE TITLE
Build from source fails

### ISSUE THREAD

**Issue by ekostem** [pytorch/pytorch#10678]
```
pytorch/aten/src/ATen/native/Convolution.cpp: In member function ‘bool at::native::ConvParams::use_mkldnn(const at::Tensor&) const’:
/home/ekostem/dev/pytorch/aten/src/ATen/native/Convolution.cpp:123:33: error: no match for ‘operator==’ (operand types are ‘at::Backend’ and ‘const at::DeviceType’)
   return input.type().backend() == kCPU &&
                                 ^
```

OS: Ubuntu 16.04.5 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.12.0

Python version: 3.6
CUDA runtime version: 9.2.148
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 396.51
cuDNN version: Probably one of the following:
/usr/local/cuda-9.2/lib64/libcudnn.so.7.1.4
/usr/local/cuda-9.2/lib64/libcudnn_static.a

**Comment by zou3519** [comment]
Could you try to pull master again and rebuild?

**Comment by ezyang** [comment]
Fixed in 0cce4620fe41c9a7ec758e6621cb9083f32d45af
